processing/src/main/java/io/druid/query/GenericQueryMetricsFactory.java

```diff
@@ -23,11 +23,11 @@
  * This factory is used for DI of custom {@link QueryMetrics} implementations for all query types, which don't (yet)
  * need to emit custom dimensions and/or metrics, i. e. they are good with the generic {@link QueryMetrics} interface.
  */
-public interface GenericQueryMetricsFactory
+public interface GenericQueryMetricsFactory<QueryType extends Query<?>>
 {
   /**
    * Creates a {@link QueryMetrics} for query, which doesn't have predefined QueryMetrics subclass. This method must
    * call {@link QueryMetrics#query(Query)} with the given query on the created QueryMetrics object before returning.
    */
-  QueryMetrics<Query<?>> makeMetrics(Query<?> query);
+  QueryMetrics<QueryType> makeMetrics(QueryType query);
```

Comment:
It doesn't make sense, `GenericQueryMetricsFactory` is not a "generic base" for other QueryMetricsFactories, it is a query metrics factory _specifically for "any" queries_.  It must be able to accept any query type.